ext/fetch/27_eventsource.js

```diff
@@ -186,165 +192,162 @@ class EventSource extends EventTarget {
+        "text/event-stream",
+      )
+    ) {
+      this[_failConnection]();
+      return;
+    }
 
-      if (this[_readyState] !== CLOSED) {
-        this[_readyState] = OPEN;
-        this.dispatchEvent(new Event("open"));
+    if (this[_readyState] === CLOSED) {
+      return;
+    }
+    this[_readyState] = OPEN;
+    this.dispatchEvent(new Event("open"));
 
-        let data = "";
-        let eventType = "";
-        let lastEventID = this[_lastEventID];
+    let data = "";
+    let eventType = "";
+    let lastEventID = this[_lastEventID];
 
-        for await (
-          // deno-lint-ignore prefer-primordials
-          const chunk of res.body.stream
-            .pipeThrough(new TextDecoderStream())
-            .pipeThrough(new TextLineStream({ allowCR: true }))
-        ) {
-          if (chunk === "") {
-            this[_lastEventID] = lastEventID;
-            if (data === "") {
-              eventType = "";
-              continue;
-            }
-            if (StringPrototypeEndsWith(data, "\n")) {
-              data = StringPrototypeSlice(data, 0, -1);
-            }
-            const event = new MessageEvent(eventType || "message", {
-              data,
-              origin: res.url(),
-              lastEventId: this[_lastEventID],
-            });
-            setIsTrusted(event, true);
-            data = "";
+    try {
+      for await (
+        // deno-lint-ignore prefer-primordials
+        const chunk of res.body.stream
+          .pipeThrough(new TextDecoderStream())
+          .pipeThrough(new TextLineStream({ allowCR: true }))
+      ) {
+        if (chunk === "") {
+          this[_lastEventID] = lastEventID;
+          if (data === "") {
             eventType = "";
-            if (this[_readyState] !== CLOSED) {
-              this.dispatchEvent(event);
-            }
-          } else if (StringPrototypeStartsWith(chunk, ":")) {
             continue;
-          } else {
-            let field = chunk;
-            let value = "";
-            if (StringPrototypeIncludes(chunk, ":")) {
-              ({ 0: field, 1: value } = StringPrototypeSplit(chunk, ":"));
-              if (StringPrototypeStartsWith(value, " ")) {
-                value = StringPrototypeSlice(value, 1);
-              }
+          }
+          if (StringPrototypeEndsWith(data, "\n")) {
+            data = StringPrototypeSlice(data, 0, -1);
+          }
+          const event = new MessageEvent(eventType || "message", {
+            data,
+            origin: res.url(),
+            lastEventId: this[_lastEventID],
+          });
+          setIsTrusted(event, true);
+          data = "";
+          eventType = "";
+          if (this[_readyState] !== CLOSED) {
+            this.dispatchEvent(event);
+          }
+        } else if (StringPrototypeStartsWith(chunk, ":")) {
+          continue;
+        } else {
+          let field = chunk;
+          let value = "";
+          if (StringPrototypeIncludes(chunk, ":")) {
```

Comment:
**nitpick:** Of course not your fault as it's in the original code but it would probably be a tad bit more performant to do `StringPrototypeIndexOf` and `StringPrototypeSubstring` to get the field and value.